frontend/test/unit/specs/pages/single-image-result-page.spec.js

```diff
@@ -0,0 +1,46 @@
+import { fireEvent } from "@testing-library/vue"
+
+import { render } from "~~/test/unit/test-utils/render"
+
+import { IMAGE } from "~/constants/media"
+import { useSingleResultStore } from "~/stores/media/single-result"
+import { useAnalytics } from "~/composables/use-analytics"
+
+import SingleImageResultPage from "~/pages/image/_id/index.vue"
+
+jest.mock("~/composables/use-analytics", () => ({
+  useAnalytics: jest.fn(),
+}))
+describe("SingleImageResultPage", () => {
+  let options
+  let singleImageStore
+  const id = "123"
+  const sendCustomEventMock = jest.fn()
+  useAnalytics.mockImplementation(() => ({
+    sendCustomEvent: sendCustomEventMock,
+  }))
```

Comment:
Ditto here. Is it better to do this and then add a `beforeEach` that resets all mocks or to initialise mocks in a `beforeEach`? I always went back and forth but was never sure which was best :thinking: Initialising it once was always confusing to me though, as the mock history leaked between test cases.

Reply:
I just wrote what worked and did not go into the details of the order - I always go back and forth on this, too.